### ISSUE TITLE
Add ability to mute metrics that are being provided that are not needed and reducing cardinality

### ISSUE THREAD

**Issue by checketts** [micrometer-metrics/micrometer#70]
Spectator has a concept of [PlaceHolders](https://github.com/Netflix/spectator/blob/master/docs/ext/placeholders.md) for reducing cardinality. 

We need a similar ability but have it contained in the MeterRegistry so that any metric being recorded can be influenced

**Comment by jkschneider** [comment]
Partially related to #17

**Comment by checketts** [comment]
# Design ideas

## Spring Config

`OFF` Ability to disable metric completely. This will ensure the metric is never registered/exposed. (Out of scope: removing metrics if config was live updated)
```
spring.metrics.http_requests: OFF
```

`MINIMAL` Disables all dimensions/tags (perhaps alias as `MERGE_DIMENSIONS` for clarity?)
```
spring.metrics.http_requests: MINIMAL
```

`FULL` Default value for metrics that aren't set. No filtering/will occur
```
spring.metrics.http_requests: FULL
```

I also want to include a `FILTER(some filtering expression)` option eventually. I think it is out of scope for this for now. This is similar to `slf4j`'s Janino integration. 

In logging, they are hierarchical by definition. So you can enable debugging for all `org.springframework` very easily.  I believe however we solve #80 could include some sort of hierarchy. But that will also be out of scope for this issue.


## Usage proposal:

Expose boolean check for if a meter should be used. This is more needed if the user wants to avoid doing work in calculating a metric. (Overloading #89 additions)
```
registry.meter("meter_name").isEnabled()

\\Alternative
registry.counter("meter_name").isEnabled()
```

Other usage won't require code changes by users. However adding filters will be allowed (and how the initial 'meter levels' will be implemented):
```
registry.addFilter(new MeterLevelFilter());
```

`MeterFilter` interface:
```
Filter.Result filter(String meterName, Iterable<Tag> tags);
```

I'll likely update this based on how it feels when I actually code it up. Any design feedback welcome.

**Comment by jkschneider** [comment]
I looked at this a bit today, and was considering an API very similar to Logback's TurboFilter:

```
Counter c = registry.counter("my.counter");
c.filter(counterState -> MeterFilter.Reply.NEUTRAL); // <1>
c.isEnabled(); // <2>
```
1. Reply with `ACCEPT`, `NEUTRAL`, or `DECLINE`
2. If the aggregate filter response is `ACCEPT` it is forced on. If `NEUTRAL`, the decision is passed to the facaded instrumentation system which may embedded filtering logic (e.g. Atlas has a meter TTL concept).

**Comment by jkschneider** [comment]
Implementation notes:
1. **Prometheus** - straightforward if we use custom collectors for all types.
2. **Dropwizard** - probably can map `MeterFilter` to Dropwizard's `MetricFilter`?
3. **Spectator** - straightforward
4. **Atlas** - BLOCKED by the fact that `AtlasRegistry` is final. If it weren't, we could subclass it and override the `newGauge`, `newTimer`, etc methods (which are currently protected) and wrap the Atlas meters in a decorator that is aware of our filters, returning the correct value for `hasExpired`.

**Comment by checketts** [comment]
Sorry for not replying earlier. I haven't begun any work on this, just documented potential config format above so we could determine what the requirements would be.

Keep in mind we want the placeholder concept to work too (but perhaps via different mechanism?)

**Comment by jkschneider** [comment]
Related to #146

**Comment by jkschneider** [comment]
Fixed with d2beebf.
